windows/src/desktop/kmshell/util/utilkmshell.pas

```diff
@@ -577,6 +578,20 @@ function GetLongFileName(const fname: string): string;
     else Result := buf;
 end;
 
+procedure GetFileNamesInDirectory(const directoryPath: string; var fileNames: TStringDynArray);
+var
+  fileName: string;
+begin
+  // Check if the directory exists
+  if TDirectory.Exists(directoryPath) then
+  begin
+    // Retrieve file names within the directory
+    fileNames := TDirectory.GetFiles(directoryPath);
+  end
+  else
+    KL.Log('Directory does not exist.');
```

Comment:
Is this unexpected? What's the failure mode here?

It's also worth noting that `KL.Log()` doesn't do anything on if `$KLOGGING` is not defined for klog.pas (and it normally is not defined). It would probably be better to be logging somewhere where we can collect diagnostics from end users.

Reply:
Ok will update my uses of KL.Log() to use something that is actually logged.